allennlp/models/ensemble/bidaf_ensemble.py

```diff
@@ -0,0 +1,111 @@
+from overrides import overrides
+import torch
+from typing import Dict, List, Any
+
+from allennlp.models.ensemble.ensemble import Ensemble
+from allennlp.models.archival import load_archive
+from allennlp.models import Model
+from allennlp.models.reading_comprehension import BidirectionalAttentionFlow
+from allennlp.common import Params
+from allennlp.data import Vocabulary
+from allennlp.training.metrics import SquadEmAndF1
+
+@Model.register("bidaf-ensemble")
+class BidafEnsemble(Ensemble):
+
+    def __init__(self,
+                 submodels: List[BidirectionalAttentionFlow]) -> None:
+        super(BidafEnsemble, self).__init__(submodels)
+
+        # Setting this propagates calls to .eval() so dropout is disabled on the submodels in evaluation
+        # and prediction.
+        self.linears = torch.nn.ModuleList(submodels)
+
+        self.submodels = submodels
+
+        self._squad_metrics = SquadEmAndF1()
+
+    @overrides
+    def forward(self,  # type: ignore
+                question: Dict[str, torch.LongTensor],
+                passage: Dict[str, torch.LongTensor],
+                span_start: torch.IntTensor = None,
+                span_end: torch.IntTensor = None,
+                metadata: List[Dict[str, Any]] = None) -> Dict[str, torch.Tensor]:
+
+        subresults = []
+        for i, submodel in enumerate(self.submodels):
+            subresults.append(self.submodels[0].forward(question, passage, span_start, span_end, metadata))
```

Comment:
Shouldn't this be `submodel.forward(...)` instead of `self.submodels[0].forward(...)`?

Reply:
Yes, sorry--leftover troubleshooting code from when I was trying to figure out why my submodels were outputting different results (because they were not in `eval` mode).  I wouldn't look over this block too closely--I have not written any tests or tried it with differing submodels.